gitlab/resource_gitlab_project_freeze_period_test.go

```diff
@@ -0,0 +1,176 @@
+package gitlab
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabProjectFreezePeriod_basic(t *testing.T) {
+	var schedule gitlab.FreezePeriod
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectFreezePeriodDestroy,
+		Steps: []resource.TestStep{
+			// Create a project and freeze period with default options
+			{
+				Config: testAccGitlabProjectFreezePeriodConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectFreezePeriodExists("gitlab_project_freeze_period.schedule", &schedule),
+					testAccCheckGitlabProjectFreezePeriodAttributes(&schedule, &testAccGitlabProjectFreezePeriodExpectedAttributes{
+						FreezeStart:  "0 23 * * 5",
+						FreezeEnd:    "0 7 * * 1",
+						CronTimezone: "UTC",
+					}),
+				),
+			},
+			// Update the freeze period to change the parameters
+			{
+				Config: testAccGitlabProjectFreezePeriodUpdateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectFreezePeriodExists("gitlab_project_freeze_period.schedule", &schedule),
+					testAccCheckGitlabProjectFreezePeriodAttributes(&schedule, &testAccGitlabProjectFreezePeriodExpectedAttributes{
+						FreezeStart:  "0 23 * * 5",
+						FreezeEnd:    "0 7 * * 3",
+						CronTimezone: "UTC",
+					}),
+				),
+			},
+			// Update the freeze period to get back to initial settings
+			{
+				Config: testAccGitlabProjectFreezePeriodConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectFreezePeriodExists("gitlab_project_freeze_period.schedule", &schedule),
+					testAccCheckGitlabProjectFreezePeriodAttributes(&schedule, &testAccGitlabProjectFreezePeriodExpectedAttributes{
+						FreezeStart:  "0 23 * * 5",
+						FreezeEnd:    "0 7 * * 1",
+						CronTimezone: "UTC",
+					}),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
Can we add a `TestAccGitlabProjectFreezePeriod_import` test?

something that would include the step:

```
{
   ResourceName:      "gitlab_project_freeze_period.schedule",
   ImportState:       true,
   ImportStateVerify: true,
}
```

Here's a few examples from other resources:

- https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_project_cluster_test.go#L103
- https://github.com/gitlabhq/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_group_label_test.go#L87

Thanks!